### PR TITLE
Deregister postfixed instances

### PR DESCRIPTION
**Pull request by ojagodzinski**

### CODE REVIEW

File: k8s/provider_test.go [hunk 1]
```diff
@@ -514,10 +517,10 @@ func TestShouldCheckMultipleServicesWithGlobalTagsCombinedAndWithPortSpecificTag
 			Tags: []string{"a", "b", "c", "instance:podName_31000"},
 		},
 		{
-			Tags: []string{"a", "b", "c", "secureConnection:true", "instance:podName_31002"},
```
Comment (AlfredBroda) [inline k8s/provider_test.go:517]: You don't have to change this if it fails locally. It doesn't in CI.

**Review by AlfredBroda — DISMISSED** [review]

File: k8s/provider.go [hunk 2]
```diff
@@ -72,6 +73,35 @@ type ServiceProvider struct {
 	Timeout time.Duration
 }
 
+// GenerateSecured generates list of postfixed Consul services for deregistration
+func (p *ServiceProvider) GenerateSecured(ctx context.Context, services []consul.ServiceInstance) []consul.ServiceInstance {
```
Comment (tomez) [inline k8s/provider.go:77]: Could you test this function?

**Review by tomez — CHANGES_REQUESTED** [review]

File: k8s/provider_test.go (hunk 1 above)
Comment (ojagodzinski) [reply k8s/provider_test.go:517]: It was failing without sorting: https://travis-ci.org/github/allegro/consul-registration-hook/builds/727683960

**Review by tomez — APPROVED** [review]

File: k8s/provider_test.go (hunk 1 above)
Comment (AlfredBroda) [reply k8s/provider_test.go:517]: Ok, let's do it this way, and get rid of the problem for good:
```go
		tagsSorted := service.Tags
		sort.Strings(tagsSorted)
		assert.Equal(t, expectedServices[i].Tags, tagsSorted)
```

**Review by AlfredBroda — CHANGES_REQUESTED** [review]

**Review by AlfredBroda — APPROVED** [review]

**Pull request merged**